### DISCUSSION TITLE
It takes too long to create a connection

### DISCUSSION THREAD

**Discussion by ethink** [duckdb/duckdb#3689] [General]
The code is as follows:
public static void main(String[] args) {
        
		 executeQuery("SELECT COUNT(*) FROM  data2");

	}
	public static void executeQuery(String sql) {

        try {
			Class.forName("org.duckdb.DuckDBDriver");
		
       
        Properties ro_prop = new Properties();
		ro_prop.setProperty("duckdb.read_only", "False");
        Long T1 = System.currentTimeMillis();
        DuckDBConnection conn = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:C:\\test\\duckdb\\test5yi2.db",ro_prop);
        
        Long T2 = System.currentTimeMillis();
        System.out.println("Create connection:" + ( T2-T1)+"ms");
        Statement stmt = conn.createStatement();
        ResultSet rs=null;
        rs=stmt.executeQuery(sql);
        rs.close();
       
        stmt.close();
        conn.close();
        } catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
    
	}
**The log is as follows:**
Create connection:15333ms

”C:\\test\\duckdb\\test5yi2.db” The file size is 20GB, and there is a table with about 1 billion pieces of data
**How can I get faster, or am I using this wrong?**

**Comment by Alex-Monahan** [comment]
Hello and thanks for the report! 

Would you be able to provide the schema of the tables in the database? If you are using any primary keys, those must be re-created during the database connection. (Persistent indexes are being actively worked on though!).

**Reply by ethink** [reply]
The code for generating the table is as follows：
public static void  addDataToTable2(String file,int rowNum) {
		
		    try
		    {
		    	long t1=System.currentTimeMillis();
		     
		    	 Properties ro_prop = new Properties();
					ro_prop.setProperty("duckdb.read_only", "False");
				DuckDBConnection conn2=null;
				  try {
						conn2 = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:"+file);
					} catch (SQLException e) {
						// TODO Auto-generated catch block
						e.printStackTrace();
					}
				Statement stmt = conn2.createStatement();
				
				stmt.execute("CREATE TABLE data2 (n_id VARCHAR, vendor_id VARCHAR, pickup_at VARCHAR,dropoff_at VARCHAR,passenger_count VARCHAR,trip_distance VARCHAR,rate_code_id VARCHAR,store_and_fwd_flag VARCHAR,pickup_location_id VARCHAR,dropoff_location_id VARCHAR,payment_type VARCHAR,fare_amount VARCHAR,extra VARCHAR,mta_tax VARCHAR,tip_amount VARCHAR,tolls_amount VARCHAR,improvement_surcharge VARCHAR,total_amount FLOAT,congestion_surcharge FLOAT)");
				DuckDBAppender appender = conn2.createAppender("main", "data2");
			
				conn2.setAutoCommit(false);
				
				stmt.execute("START TRANSACTION");
				int index=1;
				int countIndex=1;
		     
		     long t2=System.currentTimeMillis();
		    	  for(int rowIndex=0;rowIndex<rowNum;rowIndex++) {
		    		  
		    		  for(int cellIndex=0;cellIndex<1000000;cellIndex++) {
		    			  appender.beginRow();
				    	  appender.append("n_"+rowIndex+"_"+cellIndex);
				    	  appender.append("v"+cellIndex%10);
				    	  for(int i=2;i<=16;i++) {
				    		 
				    			  appender.append("s"+i);
				    		  
				    	  }
				    	  appender.append(1F);
				    	  appender.append(2F);
				    	  appender.endRow();
				    	  
				    	  if(countIndex%(1024000*2)==0) {
				    		  stmt.execute("COMMIT");
				    		  stmt.execute("START TRANSACTION");
				    		   long current=System.currentTimeMillis();
				    		  
				    		   t2=System.currentTimeMillis();
				    		   index++;
				    		 
				    	  }
				    	  countIndex++;
		    			  
		    		  }
			    	
			      
		    	  }
		    
		      
		    	  appender.close();
			      stmt.execute("COMMIT");
			      long t3=System.currentTimeMillis();
			      System.out.println("total times:"+(t3-t1));
		      stmt.close();
		      conn2.close();
		    }catch(Exception e) {
		    	e.printStackTrace();
		    }

	}
**The structure of the table is as follows:**
"CREATE TABLE data2 (n_id VARCHAR, vendor_id VARCHAR, pickup_at VARCHAR,dropoff_at VARCHAR,passenger_count VARCHAR,trip_distance VARCHAR,rate_code_id VARCHAR,store_and_fwd_flag VARCHAR,pickup_location_id VARCHAR,dropoff_location_id VARCHAR,payment_type VARCHAR,fare_amount VARCHAR,extra VARCHAR,mta_tax VARCHAR,tip_amount VARCHAR,tolls_amount VARCHAR,improvement_surcharge VARCHAR,total_amount FLOAT,congestion_surcharge FLOAT)"

**How can the modification speed up?**
